src/InodeCache.cpp

```diff
@@ -202,6 +202,12 @@ InodeCache::hash_inode(const std::string& path,
   return true;
 }
 
+#ifdef __clang__
+#  pragma clang diagnostic push
+#  if __has_warning("-Wthread-safety-analysis")
+#    pragma clang diagnostic ignored "-Wthread-safety-analysis"
```

Comment:
This appears on FreeBSD only? Any idea why? Should this be in #ifdef __FreeBSD__?

Reply:
Addressed in 4e41655. The -Wthread-safety-analysis is a by-product of -Weverything on FreeBSD currently. I did not reproduce the same behavior with macOS's Xcode toolchain.

Reply:
Any idea if this is it a reasonable warning or a false positive?

@jrosdahl: How about we change the default of [WARNINGS_AS_ERRORS](https://github.com/ccache/ccache/blob/37ab1b3e229294503e26aad9249dd6a7e6961b56/cmake/StandardWarnings.cmake#L5) to false? (true only for CI)
So many problems from uncommon compiler/OS combinations... it doesn't make much sense to fix all of them and it is blocking people in their workflows until the issues are resolved here.

Reply:
> Any idea if this is it a reasonable warning or a false positive?
> 
> @jrosdahl: How about we change the default of [WARNINGS_AS_ERRORS](https://github.com/ccache/ccache/blob/37ab1b3e229294503e26aad9249dd6a7e6961b56/cmake/StandardWarnings.cmake#L5) to false? (true only for CI)
> So many problems from uncommon compiler/OS combinations... it doesn't make much sense to fix all of them and it is blocking people in their workflows until the issues are resolved here.

In this case, it is a false-positive as clearly the two "offending" routines are simply wrappers of pthread_mutex. However this warning does seem useful to me in some way to catch some careless mistakes.

Reply:
> @jrosdahl: How about we change the default of WARNINGS_AS_ERRORS to false? (true only for CI)
So many problems from uncommon compiler/OS combinations... it doesn't make much sense to fix all of them and it is blocking people in their workflows until the issues are resolved here.

I don't understand. Isn't WARNINGS_AS_ERRORS already false for end-user builds?

Reply:
Depends on "end-user". We decided to default to true if .git exists:
https://github.com/ccache/ccache/blob/37ab1b3e229294503e26aad9249dd6a7e6961b56/cmake/StandardWarnings.cmake#L5

So one solution for such cases is to `rm -rf .git` 🤣 👎

Reply:
OK, so one problem with this is that if the user stores unpacked release source archives in a Git repository, or unpacks and builds in the top level of a Git repository, then WARNINGS_AS_ERRORS will be enabled.

> How about we change the default of WARNINGS_AS_ERRORS to false? (true only for CI)
So many problems from uncommon compiler/OS combinations... it doesn't make much sense to fix all of them and it is blocking people in their workflows until the issues are resolved here.

I fully agree that users shouldn't be blocked. That's why flags like `-Wextra -Wpedantic -Werror -Weverything` were only present in the "dev mode" in ccache 3.x. The "dev mode" has kind of been lost after moving to CMake. I still want to be blocked as a developer since I find it good not to have to wait for a GitHub build to know that something's wrong.

I have tried to address this in 533393e6ea0e31df161ab0eddccffc26b97c50fa.